test/e2e_node/e2e_service.go

```diff
@@ -208,6 +202,45 @@ func (es *e2eService) startServer(cmd *healthCheckCommand) error {
 	return fmt.Errorf("Timeout waiting for service %s", cmd)
 }
 
+func (es *e2eService) stopService(name string) error {
+	// Since node services are started with sudo, we must get the PID of the child process.
+	out, err := exec.Command("pidof", name).Output()
+	if err != nil {
+		return fmt.Errorf("error getting PID of %s: %v", name, err)
+	}
+	pid := strings.TrimSpace(string(out))
+	pidInt, err := strconv.Atoi(pid)
+	if err != nil || pidInt <= 1 {
+		return fmt.Errorf("invalid PID %q for %s: %v", pid, name, err)
+	}
+
+	// Attempt to shut down the process in a friendly manner before forcing it.
+	const (
+		timout       = 10 * time.Second
+		pollInterval = 500 * time.Millisecond
+	)
+	for _, signal := range []string{"-HUP", "-INT", "-KILL"} {
```

Comment:
Why not send `SIGTERM`? Most linux apps are expected to terminate gracefully upon receiving SIGTERM